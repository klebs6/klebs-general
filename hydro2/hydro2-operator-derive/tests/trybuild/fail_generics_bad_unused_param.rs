// ---------------- [ File: tests/trybuild/fail_generics_bad_unused_param.rs ]
// tests/trybuild/fail_generics_bad_unused_param.rs

use hydro2_operator::*;         // wherever you define #[derive(Operator)]
use hydro2_operator_derive::*;         // wherever you define #[derive(Operator)]
use named_item::*;
use named_item_derive::*;
use std::fmt::Debug;

#[derive(Debug,NamedItem,Operator)]
#[operator(
    execute="do_nothing",
    opcode="BasicOpCode::TestOp",
    input0="X" // but we won't store X or do anything with it
)]
pub struct UnusedParamOp<X: Send + Sync + Debug> {
    name: String,
    // no fields referencing X
}

impl<X: Send + Sync + Debug> UnusedParamOp<X> {
    async fn do_nothing(&self, _val: &X) -> NetResult<()> {
        // This might or might not use X, but the struct doesn't store it
        Ok(())
    }
}

fn main() {}

use std::sync::Arc;
