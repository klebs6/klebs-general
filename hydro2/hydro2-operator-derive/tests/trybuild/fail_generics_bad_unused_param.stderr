error[E0392]: type parameter `X` is never used
  --> tests/trybuild/fail_generics_bad_unused_param.rs:16:26
   |
16 | pub struct UnusedParamOp<X: Send + Sync + Debug> {
   |                          ^ unused type parameter
   |
   = help: consider removing `X`, referring to it in a field, or using a marker such as `PhantomData`
